src/client/Microsoft.Identity.Client/ManagedIdentity/CredentialProbeManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Http.Retry;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class CredentialProbeManager
+    {
+        private const string CredentialEndpoint = "http://169.254.169.254/metadata/identity/issuecredential";
```

Comment:
Decided to go with the new endpoint for the probe as it is capable of returning `IMDS` in the header. We can always swap this to any other endpoint, because all we do is get a 400 and check the header.